apiserver/facades/agent/uniter/uniter.go

```diff
@@ -1012,8 +1024,8 @@ func (u *UniterAPI) ConfigSettings(args params.Entities) (params.ConfigSettingsR
 		}
 		err = common.ErrPerm
 		if canAccess(tag) {
-			var unit *state.Unit
-			unit, err = u.getUnit(tag)
+			var unit cache.Unit
+			unit, err = u.getCacheUnit(tag)
```

Comment:
I take it this is the core of the change and the rest is just dealing with the fallout?

Reply:
Yes.